### PR TITLE
Enhance the bulk file splitting

### PR DESCRIPTION
**Pull request by wojciecholszewski-form3**
# TL;DR

PR enhances the `SplitPactBulkFile` function:

1. Suffix the output filenames with UUID (to not overwrite the target file if the descriptions are not unique)
2. Prefix the output filenames with the source file base name

## Example:

Source filename: `fx-api-ebury-gateway.json`

**Before**
```
tree /var/folders/g1/c8dzm79114xcw_5r5_979fs00000gn/T/b747eef3-b649-45c7-9742-1bdabbd244af                                                      INT ✘  1m 45s  16:04:16 
/var/folders/g1/c8dzm79114xcw_5r5_979fs00000gn/T/b747eef3-b649-45c7-9742-1bdabbd244af
├── Create FXDeal for a spot from GBP to EUR..json
├── Create FXDeal returns amount exceeds maximum error response..json
├── Create FXDeal returns currency not allowed error response..json
├── Create FXDeal returns currency not equal error response..json
├── Create FXDeal returns gateway issue error response..json
├── Create FXDeal returns invalid party ID error response..json
├── Create FXDeal returns source currency not allowed error response..json
├── Create FXDeal returns target currency not allowed error response..json
├── Create FXDeal returns type not allowed error response..json
├── Create FXDeal returns unknown error response..json
├── Fetch FX rates for a spot from GBP to EUR with Warning..json
├── Fetch FX rates for a spot from GBP to EUR..json
├── Fetch FX rates returns amount exceeds maximum error response..json
├── Fetch FX rates returns an unknown error response..json
├── Fetch FX rates returns currency not allowed error response..json
├── Fetch FX rates returns currency not equal error response..json
├── Fetch FX rates returns gateway issue error response..json
├── Fetch FX rates returns invalid party ID error response..json
├── Fetch FX rates returns source currency not allowed error response..json
├── Fetch FX rates returns target currency not allowed error response..json
└── Fetch FX rates returns type not allowed error response..json
```

**After**
```
tree /var/folders/g1/c8dzm79114xcw_5r5_979fs00000gn/T/ca945211-4f1b-407f-ad2e-4892e4aa8ccb                                                                         ✔  16:59:03 
/var/folders/g1/c8dzm79114xcw_5r5_979fs00000gn/T/ca945211-4f1b-407f-ad2e-4892e4aa8ccb
├── fx-api-ebury-gateway.Create FXDeal for a spot from GBP to EUR..22276807-242b-4bdb-857e-296b7281a3b5.json
├── fx-api-ebury-gateway.Create FXDeal returns amount exceeds maximum error response..8303184f-b6ec-4f12-8d60-7908b00e9fe1.json
├── fx-api-ebury-gateway.Create FXDeal returns currency not allowed error response..252d312a-63cf-49cd-81af-2ce05fbdd771.json
├── fx-api-ebury-gateway.Create FXDeal returns currency not equal error response..6c94e6b0-18b9-49fd-a865-b204faca3eb0.json
├── fx-api-ebury-gateway.Create FXDeal returns gateway issue error response..f97025e7-7ca7-4f8e-b008-e1ab7dd61e2c.json
├── fx-api-ebury-gateway.Create FXDeal returns invalid party ID error response..647de1b0-38c9-4fd5-a4f5-f39ad0748dc3.json
├── fx-api-ebury-gateway.Create FXDeal returns source currency not allowed error response..e438dbdd-83f9-4902-91c8-8fbdedaf785c.json
├── fx-api-ebury-gateway.Create FXDeal returns target currency not allowed error response..e3e14f23-8a11-4612-852a-ef5c3e8d9bdf.json
├── fx-api-ebury-gateway.Create FXDeal returns type not allowed error response..a6319e69-eb44-425b-a892-bbad15c7b306.json
├── fx-api-ebury-gateway.Create FXDeal returns unknown error response..87375e8a-9b70-485e-bdcf-e0e1ecd7db23.json
├── fx-api-ebury-gateway.Fetch FX rates for a spot from GBP to EUR with Warning..85636cb6-fb7a-41ea-88a7-21eaf6c77ad8.json
├── fx-api-ebury-gateway.Fetch FX rates for a spot from GBP to EUR..4fd25c81-cde4-4ddc-be6f-b1660490dbb1.json
├── fx-api-ebury-gateway.Fetch FX rates returns amount exceeds maximum error response..6b8cab9d-eea1-4717-b43d-b8dc8292f095.json
├── fx-api-ebury-gateway.Fetch FX rates returns an unknown error response..57a43fbf-715e-4898-9ae3-f3fbfd7e4509.json
├── fx-api-ebury-gateway.Fetch FX rates returns currency not allowed error response..0fc6b251-8566-499f-a675-e348e161b5e7.json
├── fx-api-ebury-gateway.Fetch FX rates returns currency not equal error response..311354f3-36e9-41db-b492-1e5e3856ecd6.json
├── fx-api-ebury-gateway.Fetch FX rates returns gateway issue error response..bc70a0e9-edff-415d-a99a-b784200c673a.json
├── fx-api-ebury-gateway.Fetch FX rates returns invalid party ID error response..e923d3fb-e066-4547-aa74-b2eea316fd05.json
├── fx-api-ebury-gateway.Fetch FX rates returns source currency not allowed error response..9b65c902-8c0e-410e-863f-90261cf5b587.json
├── fx-api-ebury-gateway.Fetch FX rates returns target currency not allowed error response..6c8f7585-1a53-483b-be1b-b060a36c063f.json
└── fx-api-ebury-gateway.Fetch FX rates returns type not allowed error response..5c78987f-3c83-43aa-96e3-ba57f3177840.json
```

# Context

## Why `Suffix the output filenames with UUID`?

The interaction's description value is taken to name the split file. Say we have not unique descriptions across interactions. If so the interaction with a duplicated name will overwrite the previously created file = we gonna miss (at least) one interaction after splitting. UUID ensures uniqueness.

## Why `Prefix the output filenames with the source file base name.`?

According to the remark in [troubleshooting](https://github.com/form3tech-oss/go-pact-testing#troubleshooting) section in `https://github.com/form3tech-oss/go-pact-testing`
> PACT tests may fail while running against bulk files (with many interactions)
> To circumvent it, you can split bulk file into smaller ones like following:

In my current task, I need to add the `Pact Provider Verification` to the `addressbook-api` (See [PR](https://github.com/form3tech/ebury-gateway/pull/270)). In my case, I split the bulk file in order to isolate the state setup part to the scope of the single interaction. Without split, I had been struggling with cross-interactions state mutation 🤯 . After splitting, every single test is wrapped with the `IntergationTest` function which runs the setUp & tearDown functions per every interaction, so it increases the level of isolation between the pact verifications. It is a WIN-WIN combo 🎉 

Splitting the bulk file and running every interaction in isolation despite a tons of benefits has one significant downside... it outputs the pact verification for every single interaction, instead of a single file per `consumer-provider` contract 😞 

So instead of `build/pact-verifications/{consumer}-{provider}.json` with example content
```json
{"success": true,"providerApplicationVersion": "v0.1.0-521-g615328d"}
```
I have a bunch of files per interaction 🤷‍♂️ :

```json
build/pact-verifications
├── fx-api-ebury-gateway.Create\ FXDeal\ for\ a\ spot\ from\ GBP\ to\ EUR..8e8e255b-a871-4320-83bd-55ff01c2ceb1.json
├── fx-api-ebury-gateway.Create\ FXDeal\ returns\ amount\ exceeds\ maximum\ error\ response..308a7bc5-d3e3-46f3-a659-5c758b18ae50.json
├── fx-api-ebury-gateway.Create\ FXDeal\ returns\ currency\ not\ allowed\ error\ response..ddefa7b4-31df-4d4d-bc0e-583118985cb3.json
├── fx-api-ebury-gateway.Create\ FXDeal\ returns\ currency\ not\ equal\ error\ response..94347044-74be-49ed-9d38-1d09849f2de5.json
├── fx-api-ebury-gateway.Create\ FXDeal\ returns\ gateway\ issue\ error\ response..792370e7-b2ce-497a-a19a-6fd90bd9a45d.json
├── fx-api-ebury-gateway.Create\ FXDeal\ returns\ invalid\ party\ ID\ error\ response..a56c622e-cc2d-4319-82e2-f43282f8a89c.json
├── fx-api-ebury-gateway.Create\ FXDeal\ returns\ source\ currency\ not\ allowed\ error\ response..35a6bbb4-e021-4b32-bf7d-f334515920d2.json
├── fx-api-ebury-gateway.Create\ FXDeal\ returns\ target\ currency\ not\ allowed\ error\ response..f0c2bf70-3fbd-4e74-a748-ec7ab5deaa4a.json
├── fx-api-ebury-gateway.Create\ FXDeal\ returns\ type\ not\ allowed\ error\ response..54f12550-138b-4c20-b806-264844fcf85a.json
├── fx-api-ebury-gateway.Create\ FXDeal\ returns\ unknown\ error\ response..a8cb958c-309c-4227-93df-10ee7e03eda5.json
├── fx-api-ebury-gateway.Fetch\ FX\ rates\ for\ a\ spot\ from\ GBP\ to\ EUR\ with\ Warning..37a21f7c-add2-4309-9d59-16d78ba0da7e.json
├── fx-api-ebury-gateway.Fetch\ FX\ rates\ for\ a\ spot\ from\ GBP\ to\ EUR..d1d0d40c-2127-4272-b2c0-6dfedbe8d1f6.json
├── fx-api-ebury-gateway.Fetch\ FX\ rates\ returns\ amount\ exceeds\ maximum\ error\ response..482c152a-bb6f-4730-81ff-a8d5e99b9281.json
├── fx-api-ebury-gateway.Fetch\ FX\ rates\ returns\ an\ unknown\ error\ response..6610a64f-1ab0-413f-b887-6c2cb3672502.json
├── fx-api-ebury-gateway.Fetch\ FX\ rates\ returns\ currency\ not\ allowed\ error\ response..c087e3ec-d6ae-4af0-bc16-87bcae74547b.json
├── fx-api-ebury-gateway.Fetch\ FX\ rates\ returns\ currency\ not\ equal\ error\ response..70df10f3-1f90-43bc-a4fc-9801c301a687.json
├── fx-api-ebury-gateway.Fetch\ FX\ rates\ returns\ gateway\ issue\ error\ response..e268c44c-eb79-41c2-8286-a444e1f4ebfe.json
├── fx-api-ebury-gateway.Fetch\ FX\ rates\ returns\ invalid\ party\ ID\ error\ response..2ce92797-ea75-4587-93b3-2f3d39c1b80f.json
├── fx-api-ebury-gateway.Fetch\ FX\ rates\ returns\ source\ currency\ not\ allowed\ error\ response..1e4db5ef-d146-4986-a56c-6a14fdf9a717.json
├── fx-api-ebury-gateway.Fetch\ FX\ rates\ returns\ target\ currency\ not\ allowed\ error\ response..9628da00-173b-4bc9-b655-d4de15c727d1.json
├── fx-api-ebury-gateway.Fetch\ FX\ rates\ returns\ type\ not\ allowed\ error\ response..5d1ee85a-4c69-452d-ae1f-053ae36a3003.json

// content is the same across all the files
{"success": (true|false),"providerApplicationVersion": "v0.1.0-521-g615328d"}
```
And now if I want to publish pact verification to the `PactBroker` I need to merge all the `{consumer}-{provider}.*.json` jsons into a single file and publish the verification once. This is why I need to prefix the split filenames so that verifications are also prefixed and I can group and merge different consumer-provider pairs verification. This is important when `Provider` has many `Consumers` and needs to validate bunch of `{consumer-A}-{provider}`, `{consumer-B}-{provider}`, `{consumer-C}-{provider}` and so on so forth.

# Extra
Besides that, I've moved the test fixtures to the go idiomatic `testdata` directory.

### CODE REVIEW

**Comment by cyplo** [comment]
@sam-io  @sata @leghissa if you could take a look

**Comment by sata** [comment]
@wojciecholszewski-form3 perhaps we can schedule a zoom for tomorrow to walk through the solution and discuss why it had to happen. Would this work for you?

**Review by sata — APPROVED** [review]
